kie-server-parent/kie-server-services/kie-server-services-task-assigning/kie-server-services-task-assigning-core/src/main/java/org/kie/server/services/taskassigning/core/model/solver/TaskHelper.java

```diff
@@ -93,16 +94,38 @@ public static int countMatchingLabels(Task task, User user, String labelName) {
     }
 
     /**
+     * Gets the list of tasks linked to a TaskOrUser.
      * @param taskOrUser a TaskOrUser instance for the evaluation.
      * @return a list with the tasks linked to the taskOrUser.
      */
-    public static List<Task> extractTaskList(TaskOrUser taskOrUser) {
-        List<Task> result = new ArrayList<>();
-        Task task = taskOrUser.getNextTask();
-        while (task != null) {
-            result.add(task);
-            task = task.getNextTask();
+    public static List<Task> extractTasks(TaskOrUser taskOrUser) {
+        return extractTasks(taskOrUser, testedTask -> true);
+    }
+
+    /**
+     * Gets the list of tasks linked to a TaskOrUser.
+     * @param taskOrUser a TaskOrUser instance for the evaluation.
+     * @param predicate a predicate for filtering the tasks that will be included in the result.
+     * @return a list with the tasks linked to the taskOrUser that verifies the filtering predicate.
+     */
+    public static List<Task> extractTasks(TaskOrUser taskOrUser, Predicate<Task> predicate) {
```

Comment:
If this method returned a `Stream`, `hasPinnedTasks(...)` could be made more efficient by using `anyMatch()`.